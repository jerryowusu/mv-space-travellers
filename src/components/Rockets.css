.rockets {
  list-style-type: none;
  display: flex;
  flex-direction: column;
  gap: 2rem;
  width: 100%;
  box-sizing: border-box;
  align-items: center;
  justify-content: center;
  padding: 0;
}

.rocket-item {
  width: 90%;
  box-sizing: border-box;
}
