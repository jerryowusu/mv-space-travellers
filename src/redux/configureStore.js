import { combineReducers, createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rocketsReducer from './rockets/rockets';

const rootReducer = combineReducers({
  rockets: rocketsReducer,
});

const initialState = {};
const middleware = [thunk];
const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware)),
);

export default store;
