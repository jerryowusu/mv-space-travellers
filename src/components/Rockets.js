import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { reserveRocket, cancelReservation } from '../redux/rockets/rockets';
import Rocket from './Rocket';
import './Rockets.css';

const Rockets = (props) => {
  const { rockets } = props;
  const dispatch = useDispatch();
  const reserveRocketAction = (id) => {
    dispatch(reserveRocket(id));
  };

  const cancelReservationAction = (id) => {
    dispatch(cancelReservation(id));
  };

  return (
    <ul className="rockets">
      {
        rockets.map((rocket) => (
          <li key={rocket.id} className="rocket-item">
            <Rocket
              id={rocket.id}
              name={rocket.name}
              image={rocket.image}
              description={rocket.description}
              reserveRocket={reserveRocketAction}
              cancelReservation={cancelReservationAction}
              reserved={rocket.reserved}
            />
          </li>
        ))
      }
    </ul>
  );
};

Rockets.propTypes = {
  rockets: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    wikipedia: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    reserved: PropTypes.bool.isRequired,
  })).isRequired,
};

export default Rockets;
