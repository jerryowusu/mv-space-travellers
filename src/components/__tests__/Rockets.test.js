import { screen, render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';
import { Provider } from 'react-redux';
import { combineReducers, createStore, applyMiddleware } from 'redux';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';
import rocketsReducer from '../../redux/rockets/rockets';
import missionsReducer from '../../redux/missions/missions';
import Rockets from '../Rockets';

const mockRocketsData = [
  {
    id: "rocket-1",
    image: "rocket-image-1",
    name: "Rocket 1",
    description: "rocket-description-1",
    reserved: false,
    wikipedia: "rocket-link-1",
  },
  {
    id: "rocket-2",
    image: "rocket-image-2",
    name: "Rocket 2",
    description: "rocket-description-2",
    reserved: false,
    wikipedia: "rocket-link-2",
  },
  {
    id: "rocket-3",
    image: "rocket-image-3",
    name: "Rocket 3",
    description: "rocket-description-3",
    reserved: false,
    wikipedia: "rocket-link-3",
  },
];

const rootReducer = combineReducers({
  rockets: rocketsReducer,
  missions: missionsReducer,
});

const store = createStore(
  rootReducer,
  applyMiddleware(thunk),
);

describe('Rockets component tests:', () => {
  it('Rockets component renders correctly', () => {
    const rocketsPage = renderer.create(
      <Provider store={store}>
        <Rockets rockets={mockRocketsData} />
      </Provider>,
    ).toJSON();
    expect(rocketsPage).toMatchSnapshot();
  });

  it('Number of rocket elements displayed should be 3', () => {
    const { container } = render(
      <Provider store={store}>
        <Rockets rockets={mockRocketsData} />
      </Provider>
    );
    const page = container.querySelector('.rockets');
    expect(page.childElementCount).toBe(3);
  });
});