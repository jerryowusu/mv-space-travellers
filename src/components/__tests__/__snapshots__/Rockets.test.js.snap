// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component tests: Rockets component renders correctly 1`] = `
<ul
  className="rockets"
>
  <li
    className="rocket-item"
  >
    <div
      className="rocket"
    >
      <img
        alt="Rocket 1"
        className="rocket-image"
        src="rocket-image-1"
      />
      <div
        className="rocket-info"
      >
        <h2
          className="rocket-name"
        >
          Rocket 1
        </h2>
        <div
          className="rocket-description-container"
        >
          <p
            className="rocket-description"
          >
            rocket-description-1
          </p>
        </div>
        <button
          className="rocket-button unclicked"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </li>
  <li
    className="rocket-item"
  >
    <div
      className="rocket"
    >
      <img
        alt="Rocket 2"
        className="rocket-image"
        src="rocket-image-2"
      />
      <div
        className="rocket-info"
      >
        <h2
          className="rocket-name"
        >
          Rocket 2
        </h2>
        <div
          className="rocket-description-container"
        >
          <p
            className="rocket-description"
          >
            rocket-description-2
          </p>
        </div>
        <button
          className="rocket-button unclicked"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </li>
  <li
    className="rocket-item"
  >
    <div
      className="rocket"
    >
      <img
        alt="Rocket 3"
        className="rocket-image"
        src="rocket-image-3"
      />
      <div
        className="rocket-info"
      >
        <h2
          className="rocket-name"
        >
          Rocket 3
        </h2>
        <div
          className="rocket-description-container"
        >
          <p
            className="rocket-description"
          >
            rocket-description-3
          </p>
        </div>
        <button
          className="rocket-button unclicked"
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </li>
</ul>
`;
